### PR TITLE
implement a lot of crowdsale tests

### PR DESCRIPTION
**Pull request by moodysalem**

### CODE REVIEW

**Comment by moodysalem** [comment]
i apparently started off an old version of master, so had to push a couple of commits to resolve conflicts. sorry about that. ready to review now

File: contracts/Bounty0xCrowdsale.sol [hunk 1]
```diff
@@ -12,41 +12,40 @@ import './AddressWhitelist.sol';
...
         msg.sender.transfer(amount);
+        OnWithdraw(msg.sender, amount);
     }
 
     // All contributions come through the fallback function
     function () payable public whenNotPaused {
         // get the current time
         uint time = currentTime();
 
-        // must be sent with some value
```
Comment (dnzdlklc) [inline contracts/Bounty0xCrowdsale.sol]: This should come back now that we've decided to remove `MINIMUM_PARTICIPATION_USD`

File: contracts/Bounty0xCrowdsale.sol [hunk 2]
```diff
@@ -56,33 +55,28 @@ contract Bounty0xCrowdsale is KnowsTime, KnowsConstants, Ownable, BntyExchangeRa
         // require we are not at the cap
         require(totalContributions.add(msg.value) <= usdToWei(HARD_CAP_USD));
 
-        bool isDuringPresale = time < WHITELIST_END_DATE;
+        // require that it's more than the minimum contribution amount
+        require(msg.value >= usdToWei(MINIMUM_PARTICIPATION_USD));
```
Comment (dnzdlklc) [inline contracts/Bounty0xCrowdsale.sol]: This will go away as discussed.

File: contracts/Bounty0xCrowdsale.sol [hunk 3]
```diff
@@ -12,90 +12,102 @@ import './AddressWhitelist.sol';
...
+        OnWithdraw(msg.sender, amount);
     }
 
     // All contributions come through the fallback function
     function () payable public whenNotPaused {
-        // get the current time
         uint time = currentTime();
 
-        // must be sent with some value
-        require(msg.value > 0);
```
Comment (dnzdlklc) [inline contracts/Bounty0xCrowdsale.sol]: We still need this

File: test/Bounty0xCrowdsale.js [hunk 4]
```diff
@@ -1,30 +1,160 @@
...
+    });
+
+    it('should not accept contributions beyond hard cap', async () => {
+      await crowdsale.setTime(limitsEndDate);
+
+      // contribute the hard cap in wei to put the crowdsale at the hard cap
+      await contribute(contributor1, hardCapWei);
+    });
+
+    it('only accepts contributions greater than the minimum amount');
```
Comment (dnzdlklc) [inline test/Bounty0xCrowdsale.js]: This will go :)

**Pull request merged**